src/cli/self_update/windows.rs

```diff
@@ -430,3 +439,18 @@ pub fn get_add_path_methods() -> Vec<PathUpdateMethod> {
 pub fn get_remove_path_methods() -> Result<Vec<PathUpdateMethod>> {
     Ok(vec![PathUpdateMethod::Windows])
 }
+
+#[cfg(test)]
+mod tests {
+    #[test]
+    fn windows_uninstall_removes_semicolon_from_path() {
+        assert_eq!(
+            "foo",
+            super::_remove_from_path(
+                r"c:\users\example\.cargo\bin;foo",
```

Comment:
Could we also have a variant which tests where that semicolon is not present?

Reply:
No. Or rather this *is* that variant: this tests that when the path shrinks so that the ; is not needed anymore, it is removed. This is the exact test that was replaced.

The reason it *looks* different is that in the other test, the intermediate state where PATH="cargo_home+bin;foo" is never explicitly spelt out - it is created by taking foo; installing cargo; then uninstalling and checking that only foo remains, not ";foo".

Because this is a more focused test, we're not also executing the installation half; thats the only difference.